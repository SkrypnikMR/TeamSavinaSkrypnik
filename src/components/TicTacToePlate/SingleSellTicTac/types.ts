export type TSingleSellTicTac = {
    id: number;
    doStep: (arg: number) => void;
    status: string;
}

export type TSingleStyledTic = {
    id?: number;
}
