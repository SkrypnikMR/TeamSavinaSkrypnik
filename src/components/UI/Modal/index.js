import ModalComponent from './ModalComponent.jsx';
import { connect } from 'react-redux';
import { changeModalVisibility } from '/src/Store/modals/actions';
import { getModalVisibilityIsOpen } from '/src/Store/modals/selectors';

const mapStateToProps = (state, ownProps) => ({
    isOpen: getModalVisibilityIsOpen(state, ownProps),
});

const mapDispatchToProps = dispatch => ({
    changeModalVisibility: payload => dispatch(changeModalVisibility(payload)),
});

export default connect(mapStateToProps, mapDispatchToProps)(ModalComponent);
