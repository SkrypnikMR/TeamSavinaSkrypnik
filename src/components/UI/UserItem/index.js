import { connect } from 'react-redux';
import UserItem from './UserItem.jsx';
import { onlineUsersEmails } from '../../../Store/chat/selectors';

const mapStateToProps = state => ({ onlineUsersEmails: onlineUsersEmails(state) });

export default connect(mapStateToProps)(UserItem);
