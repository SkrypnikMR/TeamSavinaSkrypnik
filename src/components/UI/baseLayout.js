// import styled from 'styled-components';

export const backgroundUrls = {
  light: './public/assets/images/lightTheme.jpg',
  dark: './public/assets/images/darkTheme.jpg',
};
// default color for all transparent backgrounds, use this :)
export const TRANSPARENT_BACKGROUND = 'rgba(0, 0, 0, 0.4)';
//Button
export const bgColorDefault = '#5523C1';
export const colorDefault = 'white';
export const focusColorDefault = 'black';
export const disabledColor = '#666666';
export const bgColorDisabledDefault = '#cccccc';
export const disabledBorder = '1px solid #999999';

//Input
export const bgInpColorDefault = '#F6F7FA';
export const bgFocusColorDefault = '#716f6f';
export const focusColorDefaultInput = 'white';
export const colorDefaultInput = 'black';

export const bgColorDefaultFon = 'rgba(0, 0, 0, 0.4)';
export const bgMessage = 'rgba(196, 196, 196, 0.5)';
export const textColorBlack = 'black';
