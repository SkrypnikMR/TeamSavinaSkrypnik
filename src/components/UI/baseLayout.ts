// default color for all transparent backgrounds, use this :)
export const TRANSPARENT_BACKGROUND = 'rgba(0, 0, 0, 0.4)';
//Button
export const bgColorDefault = '#23EFFC';
export const colorDefault = 'white';
export const focusColorDefault = '#26FDF0';
export const disabledColor = '#666666';
export const bgColorDisabledDefault = '#cccccc';
export const disabledBorder = '1px solid #999999';

//Input
export const bgInpColorDefault = '#F6F7FA';
export const bgFocusColorDefault = '#716f6f';
export const focusColorDefaultInput = 'white';
export const colorDefaultInput = 'black';

export const bgColorDefaultFon = 'rgba(0, 0, 0, 0.4)';
export const bgMessage = 'rgba(196, 196, 196, 0.5)';
export const textColorBlack = 'black';

//Backgrounds
export const darkGrey = '#404040';
export const midGrey = '#808080';
export const lightGray = '#C4C4C4';
export const white = '#ffffff';
export const turquous = '#23EFFC';
export const blackPic = 'rgba(0, 0, 0)';

export const colors = {
  light: {
    background: '#23EFFC',
    backgroundSecondary: '#e5fdff',
    backgroundLogReg: '#e5fdff',
    backgroundItem: '#d7d8e2',
    button: '#2B5278',
    buttonColor: '#fff',
    buttonHover: '#346392',
    input: '#D1D1D1',
    textColor: '#687780',
    buttonText: '#404040',
    placeholderColor: '#55667B',
    AddRoomBtnBtn: '#fff',
  },
  dark: {
    background: '#404040',
    backgroundSecondary: '#C4C4C4',
    backgroundLogReg: '#404040',
    backgroundItem: '#25323f',
    button: '#2B5278',
    buttonColor: '#fff',
    buttonHover: '#346392',
    input: '#242F3D',
    textColor: 'white',
    buttonText: `${bgColorDefault}`,
    placeholderColor: '#55667B',
    AddRoomBtnBtn: '#fff',
  },
};
//MainPage
export const MAIN_PAGE_BG = '#404040';

//RoomContainer
export const ROOMS_CONTAINER_BG = '#C4C4C4';

//SingleRoom
export const SINGLE_ROOM_BG = '#808080';
