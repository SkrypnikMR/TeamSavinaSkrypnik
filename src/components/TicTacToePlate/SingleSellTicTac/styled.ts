import styled from 'styled-components'; 
import { darkGrey } from '../../UI/baseLayout';
import { TSingleStyledTic } from './types';

export const StSingleSellTicTac = styled.div < TSingleStyledTic > `
    border: solid 2px ${darkGrey};
    width: 100px;
    height: 100px;
    display: flex;
    justify-content: center;
    align-items:  center;
`;
