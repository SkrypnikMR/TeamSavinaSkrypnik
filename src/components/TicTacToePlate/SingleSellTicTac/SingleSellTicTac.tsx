import React from 'react';

import { StSingleSellTicTac } from './styled';

const SingleSellTicTac = ({ id, doStep, status }) => {
    const handleClick = () => doStep(id);
    return (
        <StSingleSellTicTac id={id} onClick={handleClick}>
            {status}
        </StSingleSellTicTac>
    );
};

export default SingleSellTicTac;
