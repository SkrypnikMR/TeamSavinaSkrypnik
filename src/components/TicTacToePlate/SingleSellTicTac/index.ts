import SingleSellTicTac from './SingleSellTicTac';

export default SingleSellTicTac;
