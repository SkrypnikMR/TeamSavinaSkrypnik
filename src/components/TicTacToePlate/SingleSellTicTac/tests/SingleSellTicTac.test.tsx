import React from 'react';
import { shallow, mount } from 'enzyme';
import SingleSellTicTac from '../SingleSellTicTac';
import { CROSS, ZEROTIC } from '../../../UI/baseLayout';

describe('SingleSellTicTac', () => {
    const props = {
            id: 1,
            doStep: jest.fn(),
            status: '',
          };
    it('Should match snapshot', () => {
        const component = shallow(<SingleSellTicTac {...props}/>);
        expect(component.html()).toMatchSnapshot();
    });
    it('Should render StSingleSellTicTac ', () => {
        const component = mount(<SingleSellTicTac {...props} />);
        expect(component.find('styled__StSingleSellTicTac')).toHaveLength(1);
    });
    it('Should render img CROSS ', () => {
        props.status = 'X';
        const component = mount(<SingleSellTicTac {...props} />);
        expect(component.find('img').props().src).toBe(CROSS);
    });
    it('Should render img ZEROTIC ', () => {
        props.status = 'O';
        const component = mount(<SingleSellTicTac {...props} />);
        expect(component.find('img').props().src).toBe(ZEROTIC);
    });
    it('Should click tic-tac-toe field and call doStep ', () => {
        props.status = '';
        const component = mount(<SingleSellTicTac {...props} />);
        component.find('styled__StSingleSellTicTac').props().onClick();
        expect(props.doStep).toHaveBeenCalledWith(props.id);
    });
});
