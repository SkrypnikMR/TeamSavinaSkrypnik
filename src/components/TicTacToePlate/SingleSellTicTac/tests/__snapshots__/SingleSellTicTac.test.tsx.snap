// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleSellTicTac Should match snapshot 1`] = `"<div id=\\"1\\" class=\\"styled__StSingleSellTicTac-sc-1ruk552-0 jiqoBQ\\"><img width=\\"40px\\" src=\\"\\"/></div>"`;
